<launch>
  <arg name="fake_execution" default="false"/>

  <node pkg="sawback_manipulation" type="sawback_pick_place" name="sawback_moveitcpp" output="screen">
    <param name="arm_planning_group"          value="right_arm"/>
    <param name="gripper_planning_group"      value="hand"/>
    <param name="eef_link"                    value="right_gripper_tip"/>
    <param name="action_name"                 value="pick_place"/>
    <param name="x_offset"                    value="-0.04"/>
    <param name="y_offset"                    value="0.0"/>
    <param name="z_offset"                    value="-0.04"/>
  </node>

  <include unless="$(arg fake_execution)" file="$(find sawback_manipulation)/launch/grasping.launch"/>

  <node unless="$(arg fake_execution)" pkg="intera_interface" type="joint_trajectory_action_server.py" name="joint_trajectory_action_server"/>
  <node unless="$(arg fake_execution)" pkg="intera_interface" type="gripper_action_server.py" name="gripper_action_server"/>

  <!-- If the joints on ridgeback are required by moveit -->
  <node unless="$(arg fake_execution)" pkg="topic_tools" type="relay" name="cmd_vel_relay"
        args="/ridgeback/joint_states /robot/joint_states"/>

  <!-- No robot_state_publisher processing the sawback urdf
       it is easiest to publish a static tf between both robots -->
  <node pkg="tf2_ros" type="static_transform_publisher"
    name="mount_link_to_base" args="0 0 0.0099 0 0 0 1 mount_link base"/>

  <include file="$(find sawback_moveit_config)/launch/sawback_moveit.launch">
    <arg name="fake_execution" value="$(arg fake_execution)"/>
  </include>

  <!-- <node name="$(anon rviz)" pkg="rviz" type="rviz" respawn="false"
          args="-d $(find sawback_manipulation)/config/sawback_demo.rviz" output="screen">
    <rosparam command="load" file="$(find sawback_moveit_config)/config/kinematics.yaml"/>
  </node> -->
</launch>
